src/ModelContextProtocol.AspNetCore/Authentication/McpAuthenticationOptions.cs

```diff
@@ -0,0 +1,141 @@
+            ArgumentNullException.ThrowIfNull(value);
+            if (value.Resource == null)
+            {
+                throw new ArgumentException("The Resource property of the metadata cannot be null. A valid resource URI is required.", nameof(value));
+            }
+            
+            _resourceMetadata = value;
+            // When static metadata is set, update the provider to use it
+            _resourceMetadataProvider = _ => _resourceMetadata;
+        }
+    }    
+    
+    /// <summary>
+    /// Gets or sets a delegate that dynamically provides resource metadata based on the HTTP context.
+    /// </summary>
+    /// <remarks>
+    /// When set, this delegate will be called to generate resource metadata for each request,
+    /// allowing dynamic customization based on the caller or other contextual information.
+    /// This takes precedence over the static <see cref="ResourceMetadata"/> property.
+    /// </remarks>
+    public Func<HttpContext, ProtectedResourceMetadata>? ProtectedResourceMetadataProvider
+    {
+        get => _resourceMetadataProvider;
+        set => _resourceMetadataProvider = value;
+    }    
+    
+    /// <summary>
+    /// Sets a static resource metadata instance that will be returned for all requests.
+    /// </summary>
+    /// <param name="metadata">The static resource metadata to use.</param>
+    /// <returns>The current options instance for method chaining.</returns>
+    /// <remarks>
+    /// This is a convenience method equivalent to setting the <see cref="ResourceMetadata"/> property.
+    /// </remarks>
+    /// <exception cref="ArgumentNullException">Thrown when metadata is null.</exception>
+    /// <exception cref="ArgumentException">Thrown when the Resource property of the metadata is null.</exception>
+    public McpAuthenticationOptions UseStaticResourceMetadata(ProtectedResourceMetadata metadata)
+    {
+        ArgumentNullException.ThrowIfNull(metadata);
+        if (metadata.Resource == null)
+        {
+            throw new ArgumentException("The Resource property of the metadata cannot be null. A valid resource URI is required.", nameof(metadata));
+        }
+        
+        ResourceMetadata = metadata;
+        return this;
+    }
+
+    /// <summary>
+    /// Sets a delegate to dynamically provide resource metadata for each request.
+    /// </summary>
+    /// <param name="provider">A delegate that returns resource metadata for a given HTTP context.</param>
+    /// <returns>The current options instance for method chaining.</returns>
+    /// <remarks>
+    /// This is a convenience method equivalent to setting the <see cref="ProtectedResourceMetadataProvider"/> property.
+    /// </remarks>
+    public McpAuthenticationOptions UseDynamicResourceMetadata(Func<HttpContext, ProtectedResourceMetadata> provider)
+    {
+        ProtectedResourceMetadataProvider = provider ?? throw new ArgumentNullException(nameof(provider));
+        return this;
+    }    
```

Comment:
```suggestion
```

I think the public properties are sufficient since these appear to be no more than setters.